src/acceleration/BaseQNAcceleration.cpp

```diff
@@ -90,101 +92,18 @@ void BaseQNAcceleration::initialize(
-  _oldValues           = Eigen::VectorXd::Zero(dataSize);
-  _oldXTilde           = Eigen::VectorXd::Zero(dataSize);
-  _residuals           = Eigen::VectorXd::Zero(dataSize);
-
-  /**
-   *  make dimensions public to all procs,
-   *  last entry _dimOffsets[IntraComm::getSize()] holds the global dimension, global,n
-   */
-  std::stringstream ss;
-  if (utils::IntraComm::isParallel()) {
-    PRECICE_ASSERT(utils::IntraComm::getCommunication() != nullptr);
-    PRECICE_ASSERT(utils::IntraComm::getCommunication()->isConnected());
-
-    if (primaryDataSize <= 0) {
-      _hasNodesOnInterface = false;
-    }
-
-    /** provide vertex offset information for all processors
-     *  mesh->getVertexOffsets() provides an array that stores the number of mesh vertices on each processor
-     *  This information needs to be gathered for all meshes. To get the number of respective unknowns of a specific processor
-     *  we need to multiply the number of vertices with the dimensionality of the vector-valued data for each coupling data.
-     */
-    _dimOffsets.resize(utils::IntraComm::getSize() + 1);
-    _dimOffsetsPrimary.resize(utils::IntraComm::getSize() + 1);
-    _dimOffsets[0]        = 0;
-    _dimOffsetsPrimary[0] = 0;
-    for (size_t i = 0; i < _dimOffsets.size() - 1; i++) {
-      int accumulatedNumberOfUnknowns        = 0;
-      int accumulatedNumberOfPrimaryUnknowns = 0;
-
-      for (auto &elem : _dataIDs) {
-        const auto &offsets = cplData.at(elem)->getVertexOffsets();
-        accumulatedNumberOfUnknowns += offsets[i] * cplData.at(elem)->getDimensions();
-        if (utils::contained(elem, _primaryDataIDs)) {
-          accumulatedNumberOfPrimaryUnknowns += offsets[i] * cplData.at(elem)->getDimensions();
-        }
-      }
-      _dimOffsets[i + 1]        = accumulatedNumberOfUnknowns;
-      _dimOffsetsPrimary[i + 1] = accumulatedNumberOfPrimaryUnknowns;
-    }
-    PRECICE_DEBUG("Number of unknowns at the interface (global): {}", _dimOffsets.back());
-    if (utils::IntraComm::isPrimary()) {
-      _infostringstream << fmt::format("\n--------\n DOFs (global): {}\n offsets: {}\n", _dimOffsets.back(), _dimOffsets);
-    }
-
-    // test that the computed number of unknown per proc equals the number of primaryDataSize actually present on that proc
-    size_t unknowns = _dimOffsets[utils::IntraComm::getRank() + 1] - _dimOffsets[utils::IntraComm::getRank()];
-    PRECICE_ASSERT(dataSize == unknowns, dataSize, unknowns);
-    size_t primaryUnknowns = _dimOffsetsPrimary[utils::IntraComm::getRank() + 1] - _dimOffsetsPrimary[utils::IntraComm::getRank()];
-    PRECICE_ASSERT(primaryDataSize == primaryUnknowns, primaryDataSize, primaryUnknowns);
-  } else {
-    _infostringstream << fmt::format("\n--------\n DOFs (global): {}\n", dataSize);
-  }
-
-  // set the number of global rows in the QRFactorization.
-  _qrV.setGlobalRows(getPrimaryLSSystemRows());
-
-  _preconditioner->initialize(subVectorSizes);
```

Comment:
Again looks to me like a refactoring that we could take care of separately to make the main part more clear.